src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -42,39 +43,45 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                 {
                     throw ExceptionUtil.ParameterInvalid(nameof(exePath));
                 }
-
-                applicationFilename = ApplicationUri;
             }
             else
             {
                 // Exe path wasn't specified, get it from the entry assembly
                 exeAssembly = Assembly.GetEntryAssembly();
 
-                if (exeAssembly == null)
-                    throw new PlatformNotSupportedException();
-
-                HasEntryAssembly = true;
+                if (exeAssembly != null)
+                {
+                    HasEntryAssembly = true;
 
-                // The original .NET Framework code tried to get the local path without using Uri.
-                // If we ever find a need to do this again be careful with the logic. "file:///" is
-                // used for local paths and "file://" for UNCs. Simply removing the prefix will make
-                // local paths relative on Unix (e.g. "file:///home" will become "home" instead of
-                // "/home").
-                string configBasePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, exeAssembly.ManifestModule.Name);
-                Uri uri = new Uri(configBasePath);
+                    // The original .NET Framework code tried to get the local path without using Uri.
+                    // If we ever find a need to do this again be careful with the logic. "file:///" is
+                    // used for local paths and "file://" for UNCs. Simply removing the prefix will make
+                    // local paths relative on Unix (e.g. "file:///home" will become "home" instead of
+                    // "/home").
+                    string configBasePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, exeAssembly.ManifestModule.Name);
+                    Uri uri = new Uri(configBasePath);
 
-                if (uri.IsFile)
-                {
-                    ApplicationUri = uri.LocalPath;
-                    applicationFilename = uri.LocalPath;
+                    if (uri.IsFile)
```

Comment:
When is this condition going to be false? As far as I can tell, the above should always going to produce an a filename.

(I understand that you have not added this, but since you are cleaning this up...)

Reply:
It looks like when this code was originally written (https://github.com/dotnet/corefx/commit/7d0bf61043e61324e1cf44f099b9ecde846f2237), it was using `new Uri(exeAssembly.CodeBase)`. It was then changed by https://github.com/dotnet/corefx/pull/20488/files to use `Path.Combine(AppDomain.CurrentDomain.BaseDirectory, exeAssembly.ManifestModule.Name)`. But the `if (uri.IsFile)` was left.

I can remove the check, and add an assert instead.

FYI - @JeremyKuhne